src/GMCPAuthenticator.cpp

```diff
@@ -0,0 +1,110 @@
+/***************************************************************************
+ *   Copyright (C) 2024 by Vadim Peretokin - vperetokin@gmail.com          *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "GMCPAuthenticator.h"
+
+#include "Host.h"
+#include "ctelnet.h"
+#include <QDebug>
+
+GMCPAuthenticator::GMCPAuthenticator(Host* pHost) : mpHost(pHost) {}
```

Comment:
I thought we did it with each thing on a new line - as it makes it easier to deal understand subsequent changes:
```suggestion
GMCPAuthenticator::GMCPAuthenticator(Host* pHost)
: mpHost(pHost)
{}
```